File: 151001/Shcherba/lab2/src/main/java/by/bsuir/entities/Note.java
package by.bsuir.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "tbl_note", schema = "distcomp")
public class Note {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    @ManyToOne (cascade=CascadeType.REFRESH)
    @JoinColumn (name="tweet_id")
    private Tweet tweet;
    @Column(name = "content")
    private String content;
}

File: 151004/Haiko/RV/Models/Note.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using RV.Models;

namespace RV.Models
{
    [Table("tbl_Note")]
    public class Note
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int id { get; set; }

        [ForeignKey("Tweet")]
        [Required]
        public int tweetId { get; set; }

        [MaxLength(2048)]
        public string content { get; set; }

        public Tweet Tweet { get; set; }
    }
}
